### ISSUE TITLE
Lab 5 test case 9_03

### ISSUE THREAD

**Issue by dlyxxx** [nus-cs2030/2223-s2#473]
My test case output for customer 10 is wrong, it seems like it did not take into account the rest time of the server after 9 is done serving by 1. 

My serve for 10 seems to be triggered by the wait event for 10, and in my wait event i do not take into account the rest time of the server. I only take into account the rest time in the serve event  Would that be the issue?

<img width="1440" alt="Screenshot 2023-03-20 at 10 42 24 AM" src="https://user-images.githubusercontent.com/122346543/226234437-0ebe11c9-e55f-4e31-adbf-46402ab73db9.png">

**Comment by geraldckj** [comment]
Have this same issue, following this thread

**Comment by calvnleong** [comment]
Could this be because when you created the wait event for customer 10, the server that you took in was not updated to rest yet, hence the wait event runs

**Comment by dylanlo8** [comment]
I think before your wait gets triggered by 10, the server in the serverlist that it looks at needs to update its rest time first before 10 looks at it. Since your Done event gets triggered before wait, the updating of rest time should be there. So when wait by 10 gets called, it will see that "oh its not time to be served yet since my server is resting" -> go to continue waiting to be called again later once server has rested. hope this helps!

**Comment by tinghwei** [comment]
You need to update your server available time in Done event with the rest time accounted for and reference the updated server in Wait event instead of the server you took in when creating Wait event

**Comment by shortlion99** [comment]
> My serve for 10 seems to be triggered by the wait event for 10, and in my wait event i do not take into account the rest time of the server. I only take into account the rest time in the serve event Would that be the issue?

Yup that is the issue. restTime should be called after a server is done serving a customer, and you will have to update the server's next available time in the Done event instead of the Serve event.

**Comment by dlyxxx** [comment]
@dylanlo8 oh so you had another wait class to store the customer?

**Comment by dylanlo8** [comment]
@dlyxxx Nope, its more of checking with the MOST UPDATED server's nextavailable timing / next serve timing after the server has rested.

**Comment by kyledilao777** [comment]
@dlyxxx u can try to check your server state at the time when 10 arrived cuz by right he's supposed to leave right? can check the serverList state to see whats the issue

**Comment by markychamp** [comment]
I experienced the same issue. the problem should lie in your wait class

**Comment by haoyanggggg** [comment]
@markychamp hi can u give some hints to where in wait is the problem and how to solve it? im facing the exact same issue rn
